dbcrossbarlib/src/tokio_glue.rs

```diff
@@ -50,18 +46,83 @@ pub(crate) fn bytes_channel(
     buffer: usize,
 ) -> (
     mpsc::Sender<Result<BytesMut>>,
-    impl Stream<Item = BytesMut, Error = Error> + Send + 'static,
+    impl Stream<Item = Result<BytesMut>> + Send + Unpin + 'static,
 ) {
     let (sender, receiver) = mpsc::channel(buffer);
-    let receiver = receiver
-        // Change `Error` from `mpsc::Error` to our standard `Error`.
-        .map_err(|_| format_err!("stream read error"))
-        // Change `Item` from `Result<BytesMut>` to `BytesMut`, pushing
-        // the error into the stream's `Error` channel instead.
-        .and_then(|result| result);
```

Comment:
We no longer need to do this thing where we collapse "channel payload errors" and "channel infrastructure errors" into a single error type, because the infrastructure errors have gone away.